src/components/dataset/schemaCreation/DatasetSchemaCreationView.test.tsx

```diff
@@ -0,0 +1,1021 @@
+      //             },
+      //           ],
+      //           primaryKey: [],
+      //         },
+      //       ],
+      //       relationships: [
+      //         {
+      //           name: 'rel-name',
+      //           from: {
+      //             table: 'table_name1',
+      //             column: 'new_column1',
+      //           },
+      //           to: {
+      //             table: 'table_name',
+      //             column: 'new_column',
+      //           },
+      //         }
+      //       ],
+      //     };
+      //     expect(elem.attr('data-cy')).to.equal(JSON.stringify(comparison));
+      //   });
+      // });
+
+      it('updating table name should update relationship value', () => {
+        cy.get('#schemabuilder-createTable').click();
+        cy.get('#schemabuilder-createColumn').click();
+        cy.get('#schemabuilder-createTable').click();
+        cy.get('#schemabuilder-createColumn').click();
+        cy.get('#datasetSchema-linkRel').click();
+        cy.get('[data-cy="expand-table-button"]').eq(1).click();
+        cy.get('input[value="table_name1|column:new_column"]').eq(0).click();
+        cy.get('[data-cy="expand-table-button"]').eq(2).click();
+        cy.get('input[value="table_name|column:new_column"]').eq(0).click();
+        cy.get('#table-name').type('rel-name');
+        cy.get('#submitButton').click();
+
+        cy.get('div[data-cy="schemaBuilder-selectTableButton"] button').eq(1).click();
+        cy.get('#table-name').clear().type('party');
+
+        cy.get('.cm-theme').then((elem) => {
+          const comparison = {
+            tables: [
+              {
+                name: 'party',
+                columns: [
+                  {
+                    name: 'new_column',
+                    datatype: 'string',
+                    array_of: false,
+                    required: false,
+                  },
+                ],
+                primaryKey: [],
+              },
+              {
+                name: 'table_name1',
+                columns: [
+                  {
+                    name: 'new_column',
+                    datatype: 'string',
+                    array_of: false,
+                    required: false,
+                  },
+                ],
+                primaryKey: [],
+              },
+            ],
+            relationships: [
+              {
+                name: 'rel-name',
+                from: {
+                  table: 'table_name1',
+                  column: 'new_column',
+                },
+                to: {
+                  table: 'party',
+                  column: 'new_column',
+                },
+              },
+            ],
+          };
+          expect(elem.attr('data-cy')).to.equal(JSON.stringify(comparison));
+        });
+      });
+
+      it('updating column name should update relationship value', () => {
+        cy.get('#schemabuilder-createTable').click();
```

Comment:
The setup seems to be repeated between these tests, could that be abstracted to a helper?

Reply:
Added a helper method to call these methods!